### ISSUE TITLE
{Linked:Bug9971467} [Adaptive Cards> Designer]: Screen Reader focus is not receive consolidated focus on button and agreement label.

### ISSUE THREAD

**Issue by vagpt** [microsoft/AdaptiveCards#8186]
### Target Platforms

Other

### SDK Version

1.3

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
URL: https://adaptivecards.io/designer/
OS Version: 22H2(OS Build 25262,1000)
Browser Version: Version 110.0.1556.0 (Official build) dev (64-bit)
Screen Reader: Narrator

**Repro Steps:**
1. Open the above URL in edge browser.
2. Turn on the Narrator using 'ctrl+win+enter' key.
3. Navigate to the Designer tab item and press enter key.
4. Navigate to the 'Card Payload Editor' button and press enter key to expand it.
5. Paste the attached Json file into it.
6. Navigate to the 'Preview Mode' and press enter key.
7. Navigate to the 'Agreement label' button and observe the issue.

**Actual Result:**
While navigating the page, the screen reader focus is landing separately on agreement button and button.

**Expected Result:**
Screen Reader focus should be land entire on the agreement label and button and it should announce the state as collapsed as well.

**User Impact:**
Users would get impacted if screen reader order is not appropriate while navigating to the label and button.

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/sites/accessibility/_layouts/15/Doc.aspx?sourcedoc=%7B2F7BB36E-8B5C-4A8A-A0EA-CDDCE675D298%7D&file=MAS%201.3.2%20%E2%80%93%20Meaningful%20Sequence.docx&action=default&mobileredirect=true

"Have feedback to share on Bugs ? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0) “.

### Screenshots

https://user-images.githubusercontent.com/95913748/209625136-17fb8be1-9712-4f0d-bf23-c036446e4610.mp4

### Card JSON

```JSON
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.3",
  "body": [
    {
      "type": "Container",
      "items": [
        {
          "type": "ColumnSet",
          "columns": [
            {
              "type": "Column",
              "selectAction": {
                "type": "Action.ToggleVisibility",
                "title": "Agreement Management",
                "targetElements": [
                  "cardContent0",
                  "chevronDown0",
                  "chevronUp0"
                ]
              },
              "verticalContentAlignment": "Top",
              "items": [
                {
                  "type": "Image",
                  "id": "chevronDown0",
                  "url": "https://irisstorageaccdev.z5.web.core.windows.net/iris-ux/assets/images/Expand.png",
                  "width": "25px",
                  "altText": "Agreement Management collapsed",
                  "title": "Agreement Management"
                },
                {
                  "type": "Image",
                  "id": "chevronUp0",
                  "url": "https://irisstorageaccdev.z5.web.core.windows.net/iris-ux/assets/images/Collapse.jpg",
                  "width": "25px",
                  "altText": "Agreement Management expanded",
                  "isVisible": false,
                  "title": "Agreement Management"
                }
              ],
              "width": "auto"
            },
            {
              "type": "Column",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "**Agreement Management**",
                  "wrap": true
                },
                {
                  "type": "Container",
                  "style": "emphasis",
                  "id": "cardContent0",
                  "isVisible": false,
                  "items": [
                    {
                      "type": "ColumnSet",
                      "columns": [
                        {
                          "type": "Column",
                          "items": [
                            {
                              "type": "TextBlock",
                              "text": "agreement-management",
                              "wrap": true
                            }
                          ],
                          "width": "stretch"
                        }
                      ]
                    }
                  ]
                }
              ],
              "width": "stretch"
            }
          ]
        }
      ]
    },
    {
      "type": "Container",
      "items": [
        {
          "type": "ColumnSet",
          "columns": [
            {
              "type": "Column",
              "selectAction": {
                "type": "Action.ToggleVisibility",
                "title": "Iris",
                "targetElements": [
                  "cardContent1",
                  "chevronDown1",
                  "chevronUp1"
                ]
              },
              "verticalContentAlignment": "Top",
              "items": [
                {
                  "type": "Image",
                  "id": "chevronDown1",
                  "url": "https://irisstorageaccdev.z5.web.core.windows.net/iris-ux/assets/images/Expand.png",
                  "width": "25px",
                  "altText": "Iris collapsed",
                  "title": "Iris"
                },
                {
                  "type": "Image",
                  "id": "chevronUp1",
                  "url": "https://irisstorageaccdev.z5.web.core.windows.net/iris-ux/assets/images/Collapse.jpg",
                  "width": "25px",
                  "altText": "Iris expanded",
                  "isVisible": false,
                  "title": "Iris"
                }
              ],
              "width": "auto"
            },
            {
              "type": "Column",
              "items": [
                {
                  "type": "TextBlock",
                  "text": "**Iris**",
                  "wrap": true
                },
                {
                  "type": "Container",
                  "style": "emphasis",
                  "id": "cardContent1",
                  "isVisible": false,
                  "items": [
                    {
                      "type": "ColumnSet",
                      "columns": [
                        {
                          "type": "Column",
                          "items": [
                            {
                              "type": "TextBlock",
                              "text": "intelligent-support",
                              "wrap": true
                            }
                          ],
                          "width": "stretch"
                        }
                      ]
                    }
                  ]
                }
              ],
              "width": "stretch"
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by vagpt** [comment]
Hi @anna-dingler,

As this issue is still repro, so could you please let us know is there any future plan to fix this issue?

**Comment by anna-dingler** [comment]
Hi @vagpt,

I believe this is a card authoring issue.

For the focus problem, only the first column (icon) has `selectAction`. Since the associated text is not within this column, it will not receive focus. Both items should be contained in whichever element has the associated `selectAction` for focus to be shown properly.

For expanded/collapsed state, the card author should use `Action.ShowCard` instead of `Action.ToggleVisibility`.

Please let me know if you have any questions - thanks!

**Comment by vagpt** [comment]
Hi @anna-dingler,

Thanks for sharing the information.

We have shared your suggestion to the concern team. We will let you know whether it is working or not when receive the update from them.
